app/policies/department_policy.rb

```diff
@@ -20,31 +20,66 @@ def initialize(user, department)
     @department = department
   end
 
+  def permitted_attributes
+    return contact_attrs.push(user_ids: []) if @user.superadmin?
+    return contact_attrs if @user.department_manager?
+    []
+  end
+
   def index?
     @user.superadmin?
   end
 
   def show?
-    @user.superadmin? || @department.user.include?(@user)
+    superadmin_or_users_department?
   end
 
   def new?
     @user.superadmin?
   end
 
   def edit?
-    @user.superadmin? || @department.user.include?(@user)
+    superadmin_or_users_department?
   end
 
   def create?
     @user.superadmin?
   end
 
+  def can_associate_user?
+    @user.superadmin?
+  end
+
+  def manager_with_department?
+    @user.department_manager? && @user.with_department?
+  end
+
   def update?
-    @user.superadmin? || @department.user.include?(@user)
+    superadmin_or_users_department?
   end
 
   def destroy?
     @user.superadmin?
   end
+
+  private
+
+  def superadmin_or_users_department?
+    @user.superadmin? || @department.user.include?(@user)
```

Comment:
`@user.superadmin? || manager_with_department?` and the name of the method `superadmin_or_department_manager?`, maybe?